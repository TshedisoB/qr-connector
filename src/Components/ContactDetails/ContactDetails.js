import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import {
  ScrollView,
  View,
  Text,
  Image,
  TouchableOpacity,
  Linking,
} from "react-native";
import { NativeBaseProvider } from "native-base";
import { MaterialIcons } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";
import { Divider } from "@react-native-material/core";

import InvalidScan from "../InvalidScan/InvalidScan";
import Favorites from "../Favorites/Favorites";
import DeleteContact from "../Delete/DeleteContact";
import Notes from "../Notes/Notes";
import { setCurrentContact } from "../../Redux/actions";
import { styles } from "./ContactDetails.style";

const ContactDetails = ({ route }) => {
  const dispatch = useDispatch();
  const { barcodeData } = route.params;

  useEffect(() => {
    dispatch(setCurrentContact(barcodeData));
  }, [barcodeData]);

  if (barcodeData === "Invalid QR Code") {
    return <InvalidScan />;
  }

  const handleWebsiteLinkPress = () => {
    if (barcodeData) {
      Linking.openURL(barcodeData[0].website);
    }
  };

  const handleProfileLinkPress = () => {
    if (barcodeData) {
      Linking.openURL(barcodeData[0].profilePicLink);
    }
  };

  return (
    <NativeBaseProvider>
      <ScrollView style={styles.scrollView}>
        <LinearGradient
          colors={["purple", "maroon", "purple"]}
          style={styles.barCodeBox}>
          <View style={styles.favoriteDelete}>
            <Favorites contact={barcodeData} />
            <DeleteContact contact={barcodeData} />
          </View>

          <View style={styles.displayData}>
            <View style={styles.dataRow}>
              <MaterialIcons name="person" style={styles.icon} />
              <View style={styles.dataTextContainer}>
                <Text style={styles.dataText}>Name</Text>
                <Text style={styles.dataTextInfo}>{barcodeData[0].name}</Text>
              </View>
            </View>
            <Divider style={styles.divider} leadingInset={16} color="pink" />
            <View style={styles.dataRow}>
              <MaterialIcons name="email" style={styles.icon} />
              <View style={styles.dataTextContainer}>
                <Text style={styles.dataText}>Email</Text>
                <Text style={styles.dataTextInfo}>{barcodeData[0].email}</Text>
              </View>
            </View>
            <Divider style={styles.divider} leadingInset={16} color="pink" />
            <View style={styles.dataRow}>
              <MaterialIcons name="image" style={styles.icon} />
              <View style={styles.dataTextContainer}>
                <Text style={styles.dataText}>Profile Picture Link</Text>
                <TouchableOpacity
                  onPress={handleProfileLinkPress}
                  testID="profile-link">
                  <Image
                    source={{ uri: barcodeData[0].profilePicLink }}
                    style={styles.previewImage}
                    resizeMode="contain"
                  />
                </TouchableOpacity>
              </View>
            </View>
            <Divider style={styles.divider} leadingInset={16} color="pink" />
            <View style={styles.dataRow}>
              <MaterialIcons style={styles.icon} />
              <View style={styles.dataTextContainer}>
                <Text style={styles.dataText}>Optional Info</Text>
                <Text style={styles.linkDescription}>
                  Personal Website Link
                </Text>
                <Text
                  style={styles.dataTextInfo}
                  onPress={handleWebsiteLinkPress}
                  testID="website-link"
                  selectable={true}>
                  {barcodeData[0].website}
                </Text>
              </View>
            </View>
            <Divider style={styles.divider} leadingInset={16} color="pink" />
            <Notes contact={barcodeData} />
          </View>
        </LinearGradient>
      </ScrollView>
    </NativeBaseProvider>
  );
};

export default ContactDetails;
