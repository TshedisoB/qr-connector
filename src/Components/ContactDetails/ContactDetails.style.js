import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: "center",
    alignSelf: "center",
    height: "100%",
  },
  scrollView: {
    backgroundColor: "purple",
  },
  favoriteDelete: {
    marginTop: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginHorizontal: 50,
  },
  barCodeBox: {
    flex: 1,
    justifyContent: "center",
  },
  displayData: {
    backgroundColor: "#f8f8f8",
    borderRadius: 8,
    margin: 20,
    padding: 16,
  },
  dataRow: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 8,
  },
  icon: {
    marginRight: 8,
    color: "green",
    fontSize: 20,
  },
  dataTextContainer: {
    marginLeft: 7,
  },
  dataText: {
    fontSize: 16,
  },
  dataTextInfo: {
    fontSize: 16,
    backgroundColor: "#054405",
    color: "white",
    padding: 3,
    borderRadius: 15,
  },
  previewImage: {
    width: 70,
    height: 70,
    borderRadius: 5,
    marginVertical: 5,
  },
  linkDescription: {
    fontSize: 14,
    color: "gray",
  },
  divider: {
    margin: 5,
  },
});
