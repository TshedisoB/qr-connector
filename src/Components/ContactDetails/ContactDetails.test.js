import React from "react";
import renderer, { act } from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import { render, screen } from "@testing-library/react-native";

import ContactDetails from "./ContactDetails";
import barCodeReducer from "../../Redux/reducers";

const Stack = createStackNavigator();
const store = createStore(barCodeReducer);

jest.mock("../../sqlQueries/sqlQueries", () => ({
  __esModule: true,
  getAllContacts: jest.fn(),
  updateFavorites: jest.fn(),
  addOrUpdateNotes: jest.fn(),
}));

describe("ContactDetails component", () => {
  const barcodeData = [
    {
      name: "Tshediso Boshiana",
      email: "tshediso.boshiana@umuzi.org",
      profilePicLink: "https://github.com/tshedisoB.png",
      website: "https://tshedisob.github.io/",
      status:
        "Just discovered a hidden gem of a cafe in the heart of the city! â˜• Highly recommend trying their signature caramel latte.",
    },
  ];

  let tree;

  beforeAll(async () => {
    await act(async () => {
      tree = renderer.create(
        <Provider store={store}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="Contact Details"
                component={ContactDetails}
                initialParams={{ barcodeData }}
              />
            </Stack.Navigator>
          </NavigationContainer>
        </Provider>
      );
    });
  });

  it("should render `ContactDetails` component correctly", () => {
    act(() => {
      expect(tree).toMatchSnapshot();
    });
  });

  it("should display data on the App when it's valid", async () => {
    const name = tree.root.findAllByProps({
      children: "Tshediso Boshiana",
    });
    const email = tree.root.findAllByProps({
      children: "tshediso.boshiana@umuzi.org",
    });
    const profileLink = tree.root.findAllByProps({
      testID: "profile-link",
    });
    const webLink = tree.root.findAllByProps({
      testID: "website-link",
    });

    expect(name.length).toBeGreaterThan(0);
    expect(email.length).toBeGreaterThan(0);
    expect(profileLink.length).toBeGreaterThan(0);
    expect(webLink.length).toBeGreaterThan(0);
  });

  it("should display `No data found` when data is invalid", () => {
    const invalidData = "Invalid QR Code";

    render(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen
              name="Contact Details"
              component={ContactDetails}
              initialParams={{ barcodeData: invalidData }}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>
    );

    expect(screen.getByTestId("invalid-data")).toBeTruthy();
    expect(screen.getByText("Invalid QR code data")).toBeTruthy();
  });
});
