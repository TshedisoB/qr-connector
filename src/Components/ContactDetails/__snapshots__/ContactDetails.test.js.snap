// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails component should render \`ContactDetails\` component correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Contact Details
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaProvider
                        onInsetsChange={[Function]}
                        style={
                          [
                            {
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                      >
                        <RCTScrollView
                          style={
                            {
                              "backgroundColor": "purple",
                            }
                          }
                        >
                          <View>
                            <ViewManagerAdapter_ExpoLinearGradient
                              proxiedProperties={
                                {
                                  "colors": [
                                    4286578816,
                                    4286578688,
                                    4286578816,
                                  ],
                                  "endPoint": undefined,
                                  "locations": undefined,
                                  "startPoint": undefined,
                                }
                              }
                              style={
                                {
                                  "flex": 1,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginHorizontal": 50,
                                    "marginTop": 10,
                                  }
                                }
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "margin": 4,
                                    }
                                  }
                                >
                                  <View
                                    accessibilityState={
                                      {
                                        "busy": undefined,
                                        "checked": undefined,
                                        "disabled": undefined,
                                        "expanded": undefined,
                                        "selected": undefined,
                                      }
                                    }
                                    accessibilityValue={
                                      {
                                        "max": undefined,
                                        "min": undefined,
                                        "now": undefined,
                                        "text": undefined,
                                      }
                                    }
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      {
                                        "alignItems": "center",
                                        "backgroundColor": "#eee",
                                        "borderRadius": 5,
                                        "opacity": 1,
                                        "padding": 10,
                                      }
                                    }
                                    testID="favorite-status"
                                  >
                                    <Text
                                      style={
                                        {
                                          "marginRight": 10,
                                        }
                                      }
                                    >
                                      Add to favorites
                                    </Text>
                                    <Text
                                      allowFontScaling={false}
                                      selectable={false}
                                      style={
                                        [
                                          {
                                            "color": undefined,
                                            "fontSize": 12,
                                          },
                                          {
                                            "fontSize": 20,
                                          },
                                          {
                                            "fontFamily": "anticon",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          {},
                                        ]
                                      }
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    {
                                      "alignItems": "flex-end",
                                      "flex": 1,
                                      "marginRight": 20,
                                    }
                                  }
                                >
                                  <View
                                    accessibilityState={
                                      {
                                        "busy": undefined,
                                        "checked": undefined,
                                        "disabled": undefined,
                                        "expanded": undefined,
                                        "selected": undefined,
                                      }
                                    }
                                    accessibilityValue={
                                      {
                                        "max": undefined,
                                        "min": undefined,
                                        "now": undefined,
                                        "text": undefined,
                                      }
                                    }
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      {
                                        "alignItems": "center",
                                        "backgroundColor": "#eee",
                                        "borderRadius": 5,
                                        "opacity": 1,
                                        "padding": 10,
                                      }
                                    }
                                    testID="delete-contact"
                                  >
                                    <Text>
                                      Delete
                                    </Text>
                                    <Text>
                                      Contact
                                    </Text>
                                    <Text
                                      allowFontScaling={false}
                                      selectable={false}
                                      style={
                                        [
                                          {
                                            "color": undefined,
                                            "fontSize": 12,
                                          },
                                          {
                                            "color": "green",
                                            "fontSize": 22,
                                          },
                                          {
                                            "fontFamily": "anticon",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          {},
                                        ]
                                      }
                                    >
                                      
                                    </Text>
                                  </View>
                                  <Modal
                                    animationType="slide"
                                    hardwareAccelerated={false}
                                    onRequestClose={[Function]}
                                    transparent={true}
                                    visible={false}
                                  />
                                </View>
                              </View>
                              <View
                                style={
                                  {
                                    "backgroundColor": "#f8f8f8",
                                    "borderRadius": 8,
                                    "margin": 20,
                                    "padding": 16,
                                  }
                                }
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginBottom": 8,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    selectable={false}
                                    style={
                                      [
                                        {
                                          "color": undefined,
                                          "fontSize": 12,
                                        },
                                        {
                                          "color": "green",
                                          "fontSize": 20,
                                          "marginRight": 8,
                                        },
                                        {
                                          "fontFamily": "material",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                  <View
                                    style={
                                      {
                                        "marginLeft": 7,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        {
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      Name
                                    </Text>
                                    <Text
                                      style={
                                        {
                                          "backgroundColor": "#054405",
                                          "borderRadius": 15,
                                          "color": "white",
                                          "fontSize": 16,
                                          "padding": 3,
                                        }
                                      }
                                    >
                                      Tshediso Boshiana
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    [
                                      {
                                        "backgroundColor": "pink",
                                        "height": 1,
                                        "marginEnd": undefined,
                                        "marginStart": 16,
                                      },
                                      {
                                        "margin": 5,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginBottom": 8,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    selectable={false}
                                    style={
                                      [
                                        {
                                          "color": undefined,
                                          "fontSize": 12,
                                        },
                                        {
                                          "color": "green",
                                          "fontSize": 20,
                                          "marginRight": 8,
                                        },
                                        {
                                          "fontFamily": "material",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                  <View
                                    style={
                                      {
                                        "marginLeft": 7,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        {
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      Email
                                    </Text>
                                    <Text
                                      style={
                                        {
                                          "backgroundColor": "#054405",
                                          "borderRadius": 15,
                                          "color": "white",
                                          "fontSize": 16,
                                          "padding": 3,
                                        }
                                      }
                                    >
                                      tshediso.boshiana@umuzi.org
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    [
                                      {
                                        "backgroundColor": "pink",
                                        "height": 1,
                                        "marginEnd": undefined,
                                        "marginStart": 16,
                                      },
                                      {
                                        "margin": 5,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginBottom": 8,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    selectable={false}
                                    style={
                                      [
                                        {
                                          "color": undefined,
                                          "fontSize": 12,
                                        },
                                        {
                                          "color": "green",
                                          "fontSize": 20,
                                          "marginRight": 8,
                                        },
                                        {
                                          "fontFamily": "material",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                  <View
                                    style={
                                      {
                                        "marginLeft": 7,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        {
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      Profile Picture Link
                                    </Text>
                                    <View
                                      accessibilityState={
                                        {
                                          "busy": undefined,
                                          "checked": undefined,
                                          "disabled": undefined,
                                          "expanded": undefined,
                                          "selected": undefined,
                                        }
                                      }
                                      accessibilityValue={
                                        {
                                          "max": undefined,
                                          "min": undefined,
                                          "now": undefined,
                                          "text": undefined,
                                        }
                                      }
                                      accessible={true}
                                      collapsable={false}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        {
                                          "opacity": 1,
                                        }
                                      }
                                      testID="profile-link"
                                    >
                                      <Image
                                        resizeMode="contain"
                                        source={
                                          {
                                            "uri": "https://github.com/tshedisoB.png",
                                          }
                                        }
                                        style={
                                          {
                                            "borderRadius": 5,
                                            "height": 70,
                                            "marginVertical": 5,
                                            "width": 70,
                                          }
                                        }
                                      />
                                    </View>
                                  </View>
                                </View>
                                <View
                                  style={
                                    [
                                      {
                                        "backgroundColor": "pink",
                                        "height": 1,
                                        "marginEnd": undefined,
                                        "marginStart": 16,
                                      },
                                      {
                                        "margin": 5,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginBottom": 8,
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    selectable={false}
                                    style={
                                      [
                                        {
                                          "color": undefined,
                                          "fontSize": 12,
                                        },
                                        {
                                          "color": "green",
                                          "fontSize": 20,
                                          "marginRight": 8,
                                        },
                                        {
                                          "fontFamily": "material",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  />
                                  <View
                                    style={
                                      {
                                        "marginLeft": 7,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        {
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      Optional Info
                                    </Text>
                                    <Text
                                      style={
                                        {
                                          "color": "gray",
                                          "fontSize": 14,
                                        }
                                      }
                                    >
                                      Personal Website Link
                                    </Text>
                                    <Text
                                      onPress={[Function]}
                                      selectable={true}
                                      style={
                                        {
                                          "backgroundColor": "#054405",
                                          "borderRadius": 15,
                                          "color": "white",
                                          "fontSize": 16,
                                          "padding": 3,
                                        }
                                      }
                                      testID="website-link"
                                    >
                                      https://tshedisob.github.io/
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    [
                                      {
                                        "backgroundColor": "pink",
                                        "height": 1,
                                        "marginEnd": undefined,
                                        "marginStart": 16,
                                      },
                                      {
                                        "margin": 5,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "justifyContent": "center",
                                      "padding": 16,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      {
                                        "fontSize": 15,
                                        "marginBottom": 8,
                                      }
                                    }
                                  >
                                    Additional Contact Notes
                                  </Text>
                                  <TextInput
                                    multiline={true}
                                    onChangeText={[Function]}
                                    placeholder="You can make notes here..."
                                    style={
                                      {
                                        "borderColor": "#ccc",
                                        "borderRadius": 4,
                                        "borderWidth": 1,
                                        "marginBottom": 8,
                                        "padding": 8,
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </ViewManagerAdapter_ExpoLinearGradient>
                          </View>
                        </RCTScrollView>
                      </RNCSafeAreaProvider>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
